File: Arcade/Intro/Ruby/lineEncoding.rb
def lineEncoding(s)
    subs, aux, prev = [], "", s[0]
    
    s.size.times do |i|
        if s[i] != prev
            subs.push aux
            aux = prev = s[i]
            next
        end
        aux += prev = s[i]
    end
    subs.push aux
    
    return subs.map{|v| v.size > 1 ? v.size.to_s+v[0] : v[0]}.join
end

File: Arcade/Intro/javascript/lineEncoding.js
function lineEncoding(s) {
    let subs = [], prev = s[0], aux = "";
    
    for(let i = 0; i < s.length; i++) {
        if(s[i] != prev) {
            subs.push(aux);
            aux = prev = s[i];
            continue;
        }
        aux += prev = s[i];
    }
    subs.push(aux);
    
    return subs.map(v => ""+(v.length > 1 ? v.length : "")+v.charAt(0)).join("");
}